lib/graph3d/Graph3d.js

```diff
@@ -572,60 +471,35 @@ Graph3d.prototype._getDataPoints = function (data) {
 
   if (this.style === Graph3d.STYLE.GRID ||
     this.style === Graph3d.STYLE.SURFACE) {
-    // copy all values from the google data table to a matrix
+    // copy all values from the data table to a matrix
     // the provided values are supposed to form a grid of (x,y) positions
 
     // create two lists with all present x and y values
-    var dataX = [];
-    var dataY = [];
-    for (i = 0; i < this.getNumberOfRows(data); i++) {
-      x = data[i][this.colX] || 0;
-      y = data[i][this.colY] || 0;
-
-      if (dataX.indexOf(x) === -1) {
-        dataX.push(x);
-      }
-      if (dataY.indexOf(y) === -1) {
-        dataY.push(y);
-      }
-    }
+    var dataX = this.getPresentXValues(data);
+    var dataY = this.getPresentYValues(data);
 
     var sortNumber = function (a, b) {
```

Comment:
I can now not unsee the similarity between `getPresent[X|Y]Values()` and `DataGroup.getDistinctValues()`. I will need to DRY them!